### PR TITLE
[AIRFLOW-5748] Remove python auto-detection

### PR DESCRIPTION
**Pull request by potiuk**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-5748

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - All the public functions and the classes in the PR contain docstrings that explain what it does
  - If you implement backwards incompatible changes, please leave a note in the [Updating.md](https://github.com/apache/airflow/blob/master/UPDATING.md) so we can assign it to a appropriate release

### CODE REVIEW

File: common/_autodetect_variables.sh [hunk 1]
```diff
@@ -144,8 +119,8 @@ if [[ ! ${AUTODETECT_LONG_PYTHON_VERSION} =~ python[3]\.[0-9]+ ]]; then
     exit 1
 fi
 
-# Now set the auto-detected python version based on the auto-detected long version
-export PYTHON_VERSION=${AUTODETECT_LONG_PYTHON_VERSION#python}
+# Unless python_version is
```
Comment (ashb) [inline common/_autodetect_variables.sh]:
```suggestion
# Unless python_version is set
```
Half-written comment?

**Review by ashb — APPROVED** [review]

**Review by feluelle — APPROVED** [review]
LGTM, I read it in slack but would be nice if you could add the reason why you want to remove it to the commit and/or PR description.

File: common/_autodetect_variables.sh [hunk 2]
```diff
@@ -63,38 +63,13 @@ echo
...
-
 # IMAGE_NAME might come from DockerHub and we decide which PYTHON_VERSION to use based on it (see below)
 # In case IMAGE_NAME is not set we determine PYTHON_VERSION based on the default python on the path
 # So in virtualenvs it will use the virtualenv's PYTHON_VERSION and in DockerHub it will determine
-# PYTHHON_VERSION from the IMAGE_NAME set in the DockerHub build.
+# PYTHON_VERSION from the IMAGE_NAME set in the DockerHub build.
 # See comment above in PYTHON_VERSION - we will be able to get rid of this cumbersomness when we switch
 # to a different CI system and start pushing images to DockerHub rather than build it there.
-export BASE_IMAGE_NAME=${IMAGE_NAME:=${DOCKERHUB_USER}/${DOCKERHUB_REPO}:${AIRFLOW_CONTAINER_BRANCH_NAME}-python${AUTODETECT_PYTHON_VERSION}}
+export BASE_IMAGE_NAME=${IMAGE_NAME:=${DOCKERHUB_USER}/${DOCKERHUB_REPO}:${AIRFLOW_CONTAINER_BRANCH_NAME}-python${PYTHON_VERSION=3.5}}
```
Comment (kaxil) [inline common/_autodetect_variables.sh]:
```suggestion
export BASE_IMAGE_NAME=${IMAGE_NAME:=${DOCKERHUB_USER}/${DOCKERHUB_REPO}:${AIRFLOW_CONTAINER_BRANCH_NAME}-python${PYTHON_VERSION:=3.5}}
```

??

Comment (potiuk) [reply common/_autodetect_variables.sh]: Yeah. Good catch - but it was supposed to be :- :)

Comment (potiuk) [reply common/_autodetect_variables.sh]: I do not want to set PYTHON_VERSION in this case - just use 3.5 if not defined

**Comment by potiuk** [comment]
> LGTM, I read it in slack but would be nice if you could add the reason why you want to remove it to the commit and/or PR description.

Done in the commit message.

**Comment by feluelle** [comment]
I like that 👍

**Pull request merged**